### PR TITLE
Expose some BD Components in the Plugin API

### PR DESCRIPTION
**Pull request by QbDesu**
Steps towards implementing #909.

The Editor and FloatingWindows stuff would be a little more complicated because I don't want to directly expose Monaco specific stuff or things like that in case the editor used ever changes.

### CODE REVIEW

**Comment by QbDesu** [comment]
On further inspection exposing the Editor component as is seems reasonable. Protecting it from being patched e.g. by wrapping it in a simple wrapper function is one thing that's left to do, the other would be documenting the props it takes in.
FloatingWindows.open is kind of the same, both don't expose implementation details or internals. Just exposing both through the API shouldn't cause any issues.

File: renderer/src/modules/pluginapi.js [hunk 1]
```diff
@@ -13,6 +13,10 @@ import Logger from "common/logger";
 import Patcher from "./patcher";
 import Emotes from "../builtins/emotes/emotes";
 import ipc from "./ipc";
+import ErrorBoundary from "../ui/errorboundary";
+import * as Icons from "icons";
+import Editor from "src/ui/customcss/editor";
+import FloatingWindows from "src/ui/floatingwindows";
```
Comment (dav1312) [inline renderer/src/modules/pluginapi.js:19]: Shouldn't it be like this? Otherwise it won't work
```diff
- import Editor from "src/ui/customcss/editor";
+ import Editor from "../ui/customcss/editor";

- import FloatingWindows from "src/ui/floatingwindows";
+ import FloatingWindows from "../ui/floatingwindows";
```

Comment (Inve1951) [reply renderer/src/modules/pluginapi.js:19]: looks like it -> [webpack.config.js](https://github.com/BetterDiscord/BetterDiscord/blob/a7ab276a8164018f3aeb96d4dde559bd9c7e8660/renderer/webpack.config.js#L29)

Comment (QbDesu) [reply renderer/src/modules/pluginapi.js:19]: yeah it works, but maybe relative paths would be preferable. it was automatically imported by vscode

**Comment by QbDesu** [comment]
I think this is in a state which is usable now. I'll make a quick plugin that demonstrates all features soon-ish I guess.

**Comment by rauenzi** [comment]
The API will go through an overhaul in the future, we are not looking to add temporary exposures in the meantime.

**Pull request closed without merging**